x-pack/plugins/apm/server/lib/helpers/create_es_client/create_apm_event_client/index.ts

```diff
@@ -235,8 +292,8 @@ export class APMEventClient {
     const index = processorEventsToIndex(params.apm.events, this.indices);
 
     const requestParams = {
-      index: Array.isArray(index) ? index.join(',') : index,
       ...omit(params, 'apm'),
+      index: Array.isArray(index) ? index.join(',') : index,
```

Comment:
Should this be handled in `getRequestBase`?

Reply:
For some reason, `index` should be a flat string for `termsEnum`, but it should be an array for `msearch` 😄